### ISSUE TITLE
Possible to Execute selection rather than do?

### ISSUE THREAD

**Issue by lhebates** [poidstotal/stataRun#18]
Hi,

I am new to the stataRun plugin and am very excited about using vscode to run stata scripts. 

I am wondering if it is possible to configure the extension to "execute the selection" rather than "do", which sends the selected scripts as a temporary do file. The stata editor allows for both approaches, and I had the habit of using the "execute" much more often.

The issue with the "do" approach is that any local macros defined within an earlier part of a do-file cannot be accessed by later scripts as I execute through the script piece by piece. The script would work if run entirely, but it is not my preferred approach as I debug and write code. 

Thank you for the great work creating and maintaining this extension.

Best,
Leshui

**Comment by xxia1** [comment]
Hi Leshui, thank you for bringing this up. I am having the same issue. Have you figured out a solution? Thanks!

**Comment by lhebates** [comment]
> Hi Leshui, thank you for bringing this up. I am having the same issue. Have you figured out a solution? Thanks!

No, I haven't figured out a solution yet. Look forward to any updates/solutions.
